src/CMakeLists.txt

```diff
@@ -18,10 +18,12 @@ else()
 endif()
 
 add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+find_package(Threads REQUIRED)
 
-# Link threading if building a shared library.
+# Link threading if building a shared library, otherwise let client do it.
 if (BUILD_SHARED_LIBS)
-  find_package(Threads REQUIRED)
+  target_link_libraries(benchmark PRIVATE ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
CMake is going to pass `-lpthread` to the compiler, so it is up to the compiler on how to link it. AFAIK the compiler will prefer to just about always link pthread dynamically. I think the only case where it won't do so is if your linking pthread into a static library but I'm not sure.